e2etests/pkg/k8s/pods.go

```diff
@@ -42,13 +40,16 @@ func FRRK8sPods(cs clientset.Interface) ([]*corev1.Pod, error) {
 	return res, nil
 }
 
-func RestartFRRK8sPods(cs clientset.Interface) error {
+func RestartFRRK8sPods(cs clientset.Interface, node string) error {
```

Comment:
Ignore the wrapper part. I thought we were using this beyond the gr tests.

I'd like this to do less. So let's have one function for `RestartFRRK8sPod(cs, pod)` and a way to retrieve the pod running on a given node (if we don't have it already).

From outside it's going to be

```
pod := podForNode(node)
restartPod(pod)
```

which I think is more straightforward. Also, given that we are talking about one single pod here, the waiting logic below can simplified.

Reply:
I have removed the not need loop logic, nevertheless, everything is in the same function because I need the node name to get the pod(oldPod), then delete the pod, then need the node name to fetch the pod (newPod), compare old,new are diff and status.